engine/datamanager.go

```diff
@@ -380,31 +380,30 @@ func (dm *DataManager) GetThresholdProfile(tenant, id string, skipCache bool, tr
 }
 
 func (dm *DataManager) SetThresholdProfile(th *ThresholdProfile, withIndex bool) (err error) {
-	if err = dm.DataDB().SetThresholdProfileDrv(th); err != nil {
-		return
-	}
 	if withIndex {
-		//ToDo : update SetThresholdProfile to store indexes
-		// thdsIndexers := NewReqFilterIndexer(dm, utils.ThresholdProfilePrefix, th.Tenant)
-		// for _, fltrID := range th.FilterIDs {
-		// 	var fltr *Filter
-		// 	if fltr, err = dm.GetFilter(th.Tenant, fltrID, false, utils.NonTransactional); err != nil {
-		// 		if err == utils.ErrNotFound {
-		// 			err = fmt.Errorf("broken reference to filter: %+v for threshold: %+v", fltrID, th)
-		// 		}
-		// 		return
-		// 	}
-		// 	thdsIndexers.IndexFilters(th.ID, fltr)
-		// }
-		// if dm.DataDB().GetStorageType() == utils.REDIS {
-		// 	if err = thdsIndexers.RemoveItemFromIndex(th.ID); err != nil &&
-		// 		err.Error() != utils.ErrNotFound.Error() {
-		// 		return
-		// 	}
-		// }
-		//return thdsIndexers.StoreIndexes()
+		indexer := NewReqFilterIndexer(dm, utils.ThresholdProfilePrefix, th.Tenant)
+		//remove old ThresholdProfile indexes
+		if err = indexer.RemoveItemFromIndex(th.ID); err != nil &&
+			err.Error() != utils.ErrNotFound.Error() {
+			utils.Logger.Debug(fmt.Sprintf("RemoveItemFromIndex(setTH) Error "))
+			return
+		}
+		//Verify matching Filters for every FilterID from ThresholdProfile
+		for _, fltrID := range th.FilterIDs {
+			var fltr *Filter
+			if fltr, err = dm.GetFilter(th.Tenant, fltrID, false, utils.NonTransactional); err != nil {
+				if err == utils.ErrNotFound {
+					err = fmt.Errorf("broken reference to filter: %+v for threshold: %+v", fltrID, th)
+				}
+				return
+			}
+			indexer.IndexTPFilter(FilterToTPFilter(fltr), th.ID)
+		}
+		if err = indexer.StoreIndexes(); err != nil {
+			return
+		}
 	}
-	return
+	return dm.DataDB().SetThresholdProfileDrv(th)
```

Comment:
SetThresholdProfile needs to be before indexing so we can recover indexes later.